Feat : Added Buttons

srejitk opened the pull request:
### Added Button Component

These are the 4 types of Buttons created:

1. Primary
2. Link
3. Icon
4. FAB

![Buttons](https://user-images.githubusercontent.com/13405260/152696285-4e5cb52e-108f-4b1e-9256-56d07729ff7f.png)

![Text Icon](https://user-images.githubusercontent.com/13405260/152696413-86e8edf2-316b-47b4-88be-6a12a4283022.png)

![Icon Button](https://user-images.githubusercontent.com/13405260/152696400-fe2c8bab-4aa6-4d8b-ba62-1f3b16bed359.png)

![FAB](https://user-images.githubusercontent.com/13405260/152696392-6306c984-bd91-49fb-b00c-7f979a51a916.png)

srejitk committed:
Updated gitIgnore

srejitk committed:
feat : Added Buttons

srejitk committed:
feat : Added FAB Extended

shubhamkumar648 commented on stylesheets/components/Button/button.css:
```diff
@@ -0,0 +1,153 @@
+  }
+  .btn--primary:hover {
+    color: var(--component-blue-04);
+    background-color: white;
```
can you use variable of colour

srejitk replied:
Hey Shubham! Thanks for that, I missed it from my end. Have corrected it. 
Thanks.

shubhamkumar648 commented on stylesheets/components/Button/button.css:
```diff
@@ -0,0 +1,153 @@
+  /* Success Button */
+
+  .btn--success {
+    background-color: white;
```
same here colour variable

srejitk replied:
Corrected in the latest commit.

shubhamkumar648 commented on stylesheets/components/Button/button.css:
```diff
@@ -0,0 +1,153 @@
+  .btn--rtl {
+    flex-direction: row-reverse;
+  }
+  .btn--ltr {
```
you can also use flex flow to avoid row-reverse and row, this is short hand property

srejitk replied:
Have implemented flex-flow as suggested. This was a valuable suggestion. Will keep this in mind from next time!

shubhamkumar648 commented on stylesheets/components/Button/button.css:
```diff
@@ -0,0 +1,153 @@
+  .btn--fab {
+    position: absolute;
+    border-radius: 50%;
+    width: 56px;
```
use realtive units like rem

srejitk replied:
Corrected in the latest commit.

shubhamkumar648 reviewed (commented):
conside my comment as a sugestion ,reamaining looks good

nakulsharma15 commented on stylesheets/components/Button/button.css:
```diff
@@ -0,0 +1,153 @@
+  .btn--fab {
+    position: absolute;
+    border-radius: 50%;
+    width: 56px;
+    height: 56px;
```
You Can use rems here instead of px

srejitk replied:
Hey Nakul! Thanks, Missed it from my end. have corrected it in my latest commit.

nakulsharma15 commented on stylesheets/components/Button/button.css:
```diff
@@ -0,0 +1,153 @@
+  }
+  /* Icon Button Tonal */
+  .btn--icon--tonal {
+    background-color: #c5dcfa80;
```
can use variables for colors

srejitk replied:
Added in the latest commit.

nakulsharma15 commented on stylesheets/components/Button/button.css:
```diff
@@ -0,0 +1,153 @@
+  }
+  /* Icon Button Outline */
+  .btn--icon--outline {
+    background-color: #c5dcfa80;
```
can use varibles for colors

srejitk replied:
Added in the latest commit.

nakulsharma15 reviewed (commented):
Suggested some minor changes

srejitk committed:
Fix : Added Variables for colors

srejitk committed:
Feat : Added Small Buttons


    .btn--small {
        height: 40px;
        border-radius: 16px;
    }

    .btn--small--round {
        padding: 8px 16px;
        background-color: var(--component-orange-04);
        color: white;
    }

    .btn--small--round:hover {
        padding: 8px 16px;
        background-color: var(--component-orange-01);
        color: var(--component-orange-05);
        border: 2px solid var(--component-orange-03);
    }


    .btn--outline--primary {
        background-color: white;
        border: 2px solid #C5DCFA;
    }

    .btn--outline--primary:hover {
        background-color: var(--component-blue-01);
        border: 2px solid var(--component-blue-03);
    }

    .btn--outline--secondary {
        border: 2px solid #E0E0E0;
        background-color: white;

    }


    .btn--outline--like {
        color: var(--component-red-03);
        background: white;
        border: 2px solid var(--component-red-01);
    }

    .btn--outline--like:hover {
        color: var(--component-red-05);
        background: var(--component-red-01);
        border: 2px solid var(--component-red-03);
    }

    .text--grey {
        color: var(--component-grey-05);
    }


    .btn--link--expand {
        max-height: 48px;
        border-radius: 8px;
    }

    .btn--link--expand:hover {
        color: var(--component-blue-05);
    }

srejitk committed:
Merge branch 'dev' into feat-4-Button

srejitk committed:
Update button.css
